File: frontend/src/types/goTo.ts
export type GoToEntityType =
    | 'risk'
    | 'control'
    | 'kri'
    | 'issue'
    | 'vendor'
    | 'process'
    | 'asset'
    | 'threat';

export type GoToRecord = {
    entity_type: GoToEntityType;
    business_identifier: string | null;
    display_name: string;
    status: string;
    destination: string;
};

File: backend/app/schemas/go_to.py
from typing import Literal

from pydantic import BaseModel


class GoToRecordRead(BaseModel):
    entity_type: Literal[
        "risk",
        "control",
        "kri",
        "issue",
        "vendor",
        "process",
        "asset",
        "threat",
    ]
    business_identifier: str | None = None
    display_name: str
    status: str
    destination: str
